common/src/main/scala/org/apache/celeborn/common/rpc/netty/NettyRpcEnv.scala

```diff
@@ -61,7 +65,31 @@ class NettyRpcEnv(
   private val transportContext =
     new TransportContext(transportConf, new NettyRpcHandler(dispatcher, this))
 
-  val clientFactory = transportContext.createClientFactory()
+  private def createClientBootstraps(): java.util.List[TransportClientBootstrap] = {
+    val bootstraps = new java.util.ArrayList[TransportClientBootstrap]()
+    for {
+      secContext <- securityContext
+      clientSaslContext <- secContext.clientSaslContext
+    } {
+      if (clientSaslContext.addRegistrationBootstrap) {
+        logInfo("Add registration client bootstrap")
+        bootstraps.add(new RegistrationClientBootstrap(
+          transportConf,
+          clientSaslContext.appId,
+          clientSaslContext.saslCredentials,
+          clientSaslContext.registrationInfo))
+      } else {
+        logInfo("Add sasl client bootstrap")
+        bootstraps.add(new SaslClientBootstrap(
+          transportConf,
+          clientSaslContext.appId,
+          clientSaslContext.saslCredentials))
+      }
+    }
+    bootstraps
```

Comment:
Nit:

Both in `createClientBootstraps` and in `createServerBootstraps`, we have either 0 or 1 bootstrap (essentially an `Option`) - while the code gives the impression it can be multivalued (use of `ArrayList` for `bootstraps`, etc).


I am wondering if we want to make this more explicit.
Something like:

```
private def createClientBootstraps(): JList[TransportClientBootstrap] = {
  val bootstrapOpt = securityContext.flatMap(_.clientSaslContext.map { clientSaslContext =>
    <existing code within for>
  })
  bootstrapOpt.toList.asJava
}
```


This is a bit aesthetic change though, so I am going to leave it as a nit :-)

Reply:
I made the change.